src/util/codec/mysql/json.rs

```diff
@@ -183,6 +267,78 @@ impl Serialize for Json {
     }
 }
 
+impl PartialEq for Json {
+    fn eq(&self, right: &Json) -> bool {
+        self.cmp(right) == Ordering::Equal
+    }
+}
+
+impl PartialOrd for Json {
+    fn partial_cmp(&self, right: &Json) -> Option<Ordering> {
+        let precedence_diff = self.get_precedence() - right.get_precedence();
+        if precedence_diff == 0 {
+            return match self.get_type_code() {
+                TYPE_CODE_LITERAL => {
+                    let left_data = self.as_literal().unwrap();
+                    let right_data = right.as_literal().unwrap();
```

Comment:
ref doc:
```rust
impl<T: PartialOrd> PartialOrd for Option<T>
impl<T: PartialOrd, E: PartialOrd> PartialOrd for Result<T, E>
```

This will give you some hint for how to implement comparison neatly. 😆